src/Hosting/Hosting/src/GenericHost/GenericWebHostBuilder.cs

```diff
@@ -275,12 +276,12 @@ private void UseStartup([DynamicallyAccessedMembers(StartupLinkerOptions.Accessi
                     var actionType = typeof(Action<,>).MakeGenericType(typeof(HostBuilderContext), containerType);
 
                     // Get the private ConfigureContainer method on this type then close over the container type
-                    var configureCallback = GetType().GetMethod(nameof(ConfigureContainer), BindingFlags.NonPublic | BindingFlags.Instance)
+                    var configureCallback = typeof(GenericWebHostBuilder).GetMethod(nameof(ConfigureContainerImpl), BindingFlags.NonPublic | BindingFlags.Instance)
                                                      .MakeGenericMethod(containerType)
                                                      .CreateDelegate(actionType, this);
 
                     // _builder.ConfigureContainer<T>(ConfigureContainer);
-                    typeof(IHostBuilder).GetMethods().First(m => m.Name == nameof(IHostBuilder.ConfigureContainer))
+                    typeof(IHostBuilder).GetMethod(nameof(IHostBuilder.ConfigureContainer))
```

Comment:
Previously LINQ was used to get the first `ConfigureContainer` method for some reason. Do we know why?

I'm guessing there is only a single `ConfigureContainer` on `IHostBuilder`.

Reply:
> Previously LINQ was used to get the first ConfigureContainer method for some reason. Do we know why?

Lazy and defensive. Unless we decide to break the interface, this is "safe". 😄 

> I'm guessing there is only a single ConfigureContainer on IHostBuilder.

Correct.